### DISCUSSION TITLE
Did i mess up my Raspberry Pi kernel?

### DISCUSSION THREAD

**Discussion by nicolajkl** [christgau/wsdd#106] [Q&A]
Hello, enjoying your tool here!
Not a programmer by any means, but I have a question I can't seem to find the answer to anywhere else.

So I'm running Raspberry Pi OS buster 32-bit, and I followed your installation guide for debian buster/ubuntu, by adding the thirdparty repo you list there + gpg key. (deb https://pkg.ltec.ch/public/ buster main)

So all good, it works great. But I'm a bit scared that when i did the "apt update/apt upgrade" there was a kernel update all of a sudden! So i guess this was from the repo you listed..

Question: How will that affect my Raspberry Pi OS installation? I mean everything seems to be running fine, but i got so worried about all those "kernel hacks" being listed while installing the update.

I hope you can calm my nerves a bit.
The tool and everything else seems to be working though.

Thanks again!.
![billede](https://user-images.githubusercontent.com/434582/118039720-ef740800-b370-11eb-9a1f-ca936965634f.png)

**Comment by christgau** [comment]
wsdd does not depend on any special kernel version and - as far as I can tell from the package files - the community-provided package does not list any kernel version as explicit dependency. The only dependency listed in INFO is python3. Further, wsdd does not perform any special interaction with the kernel. 

>  i got so worried about all those "kernel hacks" being listed while installing the update.

I am not sure what do you mean by "kernel hacks", but I assume you rather encountered a regular update of the kernel. You may could have verified this by doing an `apt upgrade` + `update` before you installed wsdd and the additional repository. I am pretty sure that you would have seen the kernel updates in that case as well.

**Comment by bupar01** [comment]
Steffen,

The script has been very useful for me and I have it successfully installed in several Raspberry Pis of different vintages. 

I have put together simple instructions for setting up the script as a system service for people who do not want to mess around with apt sources. You are free to include it or refer to it in your README.md.

[WSDD Installation procedure for Raspberry Pi](https://github.com/bupar01/raspberrypi-wsdd-install)

**Reply by JedMeister** [reply]
FYI it may not be supported on older versions of systems, but in 10/Buster (Debian and/or Raspbian) you can both enable and start a service with the `--now` switch. I.e. the below command would both enable and start the `wsdd.service`:

```
sudo systemctl enable --now wsdd.service
```

I'm assuming that your instructions are intended to also support older releases, so I don't necessarily recommend changing your instructions unless you check and confirm it works in older releases. But it might be useful for your own knowledge?

Also for bonus points, this also applies to disabling and stopping a service (i.,e. use the `--now` switch to both stop and disable a service).

**Reply by christgau** [reply]
> I have put together simple instructions for setting up the script as a system service for people who do not want to mess around with apt sources. 

I am not sure if I would like this approach of installing a third-party software. With packages you have a (hopefully) clean way of installing _and_ removing a software in the distro-native way. Sure, you can always put software on a system in any other way, but if you have packages I would always go this way. And the question for me would be: What's the problem with adding new packages sources? Anyways...

> [WSDD Installation procedure for Raspberry Pi](https://github.com/bupar01/raspberrypi-wsdd-install)

Would you mind distilling a script out of those instruction? That might be even more helpful for users, but in the end that would do the things which the package would do...

And one question to your service file:

> [Service]
> ExecStartPre=/bin/sleep 5

Why this?

**Reply by bupar01** [reply]
> I am not sure if I would like this approach of installing a third-party software. With packages you have a (hopefully) clean way of installing and removing a software in the distro-native way.

Point taken. The OP voiced concern regarding non-official Raspbian repos which, I think, is not uncommon amongst folks who do not have the time or expertise to examine what is in those repos.

I have two reasons for doing it this way:
1. I am very much an amateur tinkler in the process of learning how linux works. This is a learning exercise for me about systemctl services.
2. I think your implementation will be incorporated into linux distros or samba eventually and this is a short lived temporary fix. Reversing it is relatively simple - disabling the service and removing the two files, although not as clean as you mentioned in the distro-native way.

> Would you mind distilling a script out of those instruction? That might be even more helpful for users, but in the end that would do the things which the package would do...

Agree it is a duplication.

> [Service]
> ExecStartPre=/bin/sleep 5

I ended up with the service stuck in starting state without this.

The comments are appreciated. Thanks.
